File: platforms/web/src/theme/resonanceTheme.ts
/** Resonance UX Design System - Luminous Cosmic Architecture Theme Tokens */

export const colors = {
  /** Deep forest greens */
  forest: {
    900: '#0A1C14',
    800: '#122E21',
    700: '#1B402E',
    600: '#24523B',
    500: '#2D6449',
  },

  /** Gold accents */
  gold: {
    900: '#6B5420',
    800: '#9A7A3A',
    700: '#B8924A',
    600: '#C5A059',
    500: '#D4B36E',
    400: '#E6D0A1',
    300: '#F0E4C6',
    200: '#F7F0DF',
  },

  /** Cream base */
  cream: {
    100: '#FAFAF8',
    200: '#F5F4EE',
    300: '#EDECE4',
    400: '#E2E0D6',
  },

  /** Muted greens for text */
  sage: {
    700: '#3D5247',
    600: '#5C7065',
    500: '#6E8577',
    400: '#8A9C91',
    300: '#A8B8AD',
    200: '#C5D0C9',
  },

  /** Night mode */
  night: {
    900: '#030907',
    800: '#05100B',
    700: '#0A1C14',
    600: '#0F2A1E',
  },

  /** Semantic */
  error: '#B54B4B',
  success: '#5E8B5E',
  warning: '#C5A059',
  info: '#5B7B8A',
} as const;

export const typography = {
  fontFamily: {
    heading: "'Cormorant Garamond', 'Georgia', serif",
    body: "'Manrope', 'Helvetica Neue', sans-serif",
  },

  fontSize: {
    xs: '0.75rem',     // 12px
    sm: '0.875rem',    // 14px
    base: '1rem',      // 16px
    md: '1.125rem',    // 18px
    lg: '1.25rem',     // 20px
    xl: '1.5rem',      // 24px
    '2xl': '1.875rem', // 30px
    '3xl': '2.25rem',  // 36px
    '4xl': '3rem',     // 48px
    '5xl': '3.75rem',  // 60px
  },

  fontWeight: {
    light: 300,
    regular: 400,
    medium: 500,
    semibold: 600,
    bold: 700,
  },

  lineHeight: {
    tight: 1.1,
    snug: 1.3,
    normal: 1.5,
    relaxed: 1.7,
    loose: 2,
  },

  letterSpacing: {
    tight: '-0.02em',
    normal: '0',
    wide: '0.05em',
    wider: '0.1em',
    widest: '0.2em',
  },
} as const;

export const spacing = {
  0: '0',
  1: '0.25rem',
  2: '0.5rem',
  3: '0.75rem',
  4: '1rem',
  5: '1.25rem',
  6: '1.5rem',
  8: '2rem',
  10: '2.5rem',
  12: '3rem',
  16: '4rem',
  20: '5rem',
  24: '6rem',
  32: '8rem',
} as const;

export const borderRadius = {
  sm: '0.375rem',
  md: '0.75rem',
  lg: '1rem',
  xl: '1.5rem',
  '2xl': '2rem',
  full: '9999px',
} as const;

export const shadows = {
  sm: '0 1px 3px rgba(154, 122, 58, 0.08)',
  md: '0 4px 12px rgba(154, 122, 58, 0.12)',
  lg: '0 8px 24px rgba(154, 122, 58, 0.12)',
  xl: '0 12px 40px rgba(154, 122, 58, 0.16)',
  glow: '0 0 20px rgba(197, 160, 89, 0.2)',
  inner: 'inset 0 2px 8px rgba(10, 28, 20, 0.08)',
} as const;

export const glassmorphism = {
  background: {
    light: 'rgba(250, 250, 248, 0.7)',
    dark: 'rgba(10, 28, 20, 0.6)',
  },
  backdropBlur: 'blur(12px)',
  border: {
    light: '1px solid rgba(138, 156, 145, 0.25)',
    dark: '1px solid rgba(138, 156, 145, 0.15)',
  },
} as const;

export const animation = {
  easing: {
    spring: 'cubic-bezier(0.34, 1.56, 0.64, 1)',
    smooth: 'cubic-bezier(0.4, 0, 0.2, 1)',
    decelerate: 'cubic-bezier(0, 0, 0.2, 1)',
    accelerate: 'cubic-bezier(0.4, 0, 1, 1)',
  },
  duration: {
    instant: '100ms',
    fast: '200ms',
    normal: '350ms',
    slow: '500ms',
    glacial: '800ms',
    breathing: '8s',
  },
} as const;

export const breakpoints = {
  sm: '640px',
  md: '768px',
  lg: '1024px',
  xl: '1280px',
  '2xl': '1536px',
} as const;

export const zIndex = {
  background: -1,
  base: 0,
  elevated: 10,
  sticky: 20,
  overlay: 30,
  modal: 40,
  toast: 50,
  max: 100,
} as const;

const resonanceTheme = {
  colors,
  typography,
  spacing,
  borderRadius,
  shadows,
  glassmorphism,
  animation,
  breakpoints,
  zIndex,
} as const;

export type ResonanceTheme = typeof resonanceTheme;
export default resonanceTheme;

File: platforms/ios/LuminousCosmic/Sources/Theme/ResonanceTheme.swift
// ResonanceTheme.swift
// Luminous Cosmic Architecture™
// Complete Resonance UX Design System for iOS

import SwiftUI

// MARK: - Color Palette

struct ResonanceColors {
    // Deep Forest Greens
    static let forestDeep = Color(hex: "0A1C14")
    static let forestMid = Color(hex: "122E21")
    static let forestLight = Color(hex: "1B402E")

    // Gold Accents
    static let goldPrimary = Color(hex: "C5A059")
    static let goldLight = Color(hex: "E6D0A1")
    static let goldDark = Color(hex: "9A7A3A")

    // Cream Base
    static let creamPrimary = Color(hex: "FAFAF8")
    static let creamWarm = Color(hex: "F5F4EE")

    // Muted Greens (Text)
    static let textMutedDark = Color(hex: "5C7065")
    static let textMutedLight = Color(hex: "8A9C91")

    // Night Mode
    static let nightDeep = Color(hex: "05100B")
    static let nightMid = Color(hex: "0A1C14")

    // Borders
    static let borderLight = Color(hex: "8A9C91").opacity(0.25)
    static let borderGold = Color(hex: "C5A059").opacity(0.3)

    // Shadows
    static let shadowGold = Color(hex: "C5A059").opacity(0.15)
    static let shadowDark = Color.black.opacity(0.2)

    // Semantic Colors
    static let fire = Color(hex: "C5604A")
    static let earth = Color(hex: "8B7D5C")
    static let air = Color(hex: "7BA5A0")
    static let water = Color(hex: "5C7D9E")
}

// MARK: - Adaptive Theme

struct ResonanceTheme {
    let isDark: Bool

    var background: Color { isDark ? ResonanceColors.nightDeep : ResonanceColors.creamPrimary }
    var backgroundSecondary: Color { isDark ? ResonanceColors.nightMid : ResonanceColors.creamWarm }
    var surface: Color { isDark ? ResonanceColors.forestDeep : .white }
    var surfaceElevated: Color { isDark ? ResonanceColors.forestMid : ResonanceColors.creamWarm }

    var textPrimary: Color { isDark ? ResonanceColors.creamPrimary : ResonanceColors.forestDeep }
    var textSecondary: Color { isDark ? ResonanceColors.textMutedLight : ResonanceColors.textMutedDark }
    var textTertiary: Color { isDark ? ResonanceColors.textMutedDark : ResonanceColors.textMutedLight }

    var accent: Color { ResonanceColors.goldPrimary }
    var accentLight: Color { ResonanceColors.goldLight }
    var accentDark: Color { ResonanceColors.goldDark }

    var border: Color { isDark ? ResonanceColors.borderGold : ResonanceColors.borderLight }

    var glassFill: Color {
        isDark
            ? ResonanceColors.forestMid.opacity(0.4)
            : Color.white.opacity(0.45)
    }

    var glassStroke: Color {
        isDark
            ? ResonanceColors.goldPrimary.opacity(0.15)
            : ResonanceColors.textMutedLight.opacity(0.25)
    }

    // Gradients
    var backgroundGradient: LinearGradient {
        LinearGradient(
            colors: isDark
                ? [ResonanceColors.nightDeep, ResonanceColors.forestDeep, ResonanceColors.nightMid]
                : [ResonanceColors.creamPrimary, ResonanceColors.creamWarm, ResonanceColors.creamPrimary],
            startPoint: .topLeading,
            endPoint: .bottomTrailing
        )
    }

    var goldGradient: LinearGradient {
        LinearGradient(
            colors: [ResonanceColors.goldDark, ResonanceColors.goldPrimary, ResonanceColors.goldLight],
            startPoint: .topLeading,
            endPoint: .bottomTrailing
        )
    }

    var forestGradient: LinearGradient {
        LinearGradient(
            colors: [ResonanceColors.forestDeep, ResonanceColors.forestMid, ResonanceColors.forestLight],
            startPoint: .top,
            endPoint: .bottom
        )
    }

    var cardGradient: LinearGradient {
        LinearGradient(
            colors: isDark
                ? [ResonanceColors.forestMid.opacity(0.6), ResonanceColors.forestDeep.opacity(0.4)]
                : [Color.white.opacity(0.8), ResonanceColors.creamWarm.opacity(0.6)],
            startPoint: .topLeading,
            endPoint: .bottomTrailing
        )
    }

    var cosmicGradient: RadialGradient {
        RadialGradient(
            colors: isDark
                ? [ResonanceColors.forestLight.opacity(0.3), ResonanceColors.nightDeep]
                : [ResonanceColors.goldLight.opacity(0.2), ResonanceColors.creamPrimary],
            center: .center,
            startRadius: 50,
            endRadius: 400
        )
    }
}

// MARK: - Typography

struct ResonanceTypography {
    // Serif Headers (Cormorant Garamond style - using system serif as fallback)
    static func serifDisplay(_ size: CGFloat) -> Font {
        .custom("Cormorant Garamond", size: size, relativeTo: .largeTitle)
    }

    static func serifTitle(_ size: CGFloat) -> Font {
        .custom("CormorantGaramond-Medium", size: size, relativeTo: .title)
    }

    static func serifBody(_ size: CGFloat) -> Font {
        .custom("CormorantGaramond-Regular", size: size, relativeTo: .body)
    }

    // Sans-serif Body (Manrope style - using system sans as fallback)
    static func sansBody(_ size: CGFloat) -> Font {
        .custom("Manrope-Regular", size: size, relativeTo: .body)
    }

    static func sansMedium(_ size: CGFloat) -> Font {
        .custom("Manrope-Medium", size: size, relativeTo: .body)
    }

    static func sansBold(_ size: CGFloat) -> Font {
        .custom("Manrope-Bold", size: size, relativeTo: .body)
    }

    // Fallback-safe tokens
    static let displayLarge: Font = .system(size: 40, weight: .light, design: .serif)
    static let displayMedium: Font = .system(size: 32, weight: .light, design: .serif)
    static let displaySmall: Font = .system(size: 28, weight: .regular, design: .serif)
    static let headlineLarge: Font = .system(size: 24, weight: .medium, design: .serif)
    static let headlineMedium: Font = .system(size: 20, weight: .medium, design: .serif)
    static let headlineSmall: Font = .system(size: 17, weight: .semibold, design: .serif)
    static let bodyLarge: Font = .system(size: 17, weight: .regular, design: .default)
    static let bodyMedium: Font = .system(size: 15, weight: .regular, design: .default)
    static let bodySmall: Font = .system(size: 13, weight: .regular, design: .default)
    static let caption: Font = .system(size: 12, weight: .medium, design: .default)
    static let overline: Font = .system(size: 11, weight: .semibold, design: .default)
}

// MARK: - Spacing

struct ResonanceSpacing {
    static let xxxs: CGFloat = 2
    static let xxs: CGFloat = 4
    static let xs: CGFloat = 8
    static let sm: CGFloat = 12
    static let md: CGFloat = 16
    static let lg: CGFloat = 24
    static let xl: CGFloat = 32
    static let xxl: CGFloat = 48
    static let xxxl: CGFloat = 64
}

// MARK: - Corner Radii

struct ResonanceRadius {
    static let sm: CGFloat = 8
    static let md: CGFloat = 12
    static let lg: CGFloat = 16
    static let xl: CGFloat = 24
    static let pill: CGFloat = 100
}

// MARK: - Animation

struct ResonanceAnimation {
    // Spring easing matching cubic-bezier(0.34, 1.56, 0.64, 1)
    static let springBouncy = Animation.spring(response: 0.5, dampingFraction: 0.6, blendDuration: 0.3)
    static let springSmooth = Animation.spring(response: 0.4, dampingFraction: 0.8, blendDuration: 0.2)
    static let springGentle = Animation.spring(response: 0.6, dampingFraction: 0.75, blendDuration: 0.4)
    static let easeOut = Animation.easeOut(duration: 0.35)
    static let slowReveal = Animation.easeInOut(duration: 0.8)
    static let celestial = Animation.easeInOut(duration: 2.0).repeatForever(autoreverses: true)
    static let breathe = Animation.easeInOut(duration: 4.0).repeatForever(autoreverses: true)
    static let orbit = Animation.linear(duration: 60).repeatForever(autoreverses: false)
}

// MARK: - Environment Key

struct ThemeKey: EnvironmentKey {
    static let defaultValue = ResonanceTheme(isDark: false)
}

extension EnvironmentValues {
    var resonanceTheme: ResonanceTheme {
        get { self[ThemeKey.self] }
        set { self[ThemeKey.self] = newValue }
    }
}

// MARK: - Color Hex Extension

extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3:
            (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6:
            (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8:
            (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default:
            (a, r, g, b) = (255, 0, 0, 0)
        }
        self.init(
            .sRGB,
            red: Double(r) / 255,
            green: Double(g) / 255,
            blue: Double(b) / 255,
            opacity: Double(a) / 255
        )
    }
}

// MARK: - Haptics

struct ResonanceHaptics {
    static func light() {
        UIImpactFeedbackGenerator(style: .light).impactOccurred()
    }

    static func medium() {
        UIImpactFeedbackGenerator(style: .medium).impactOccurred()
    }

    static func soft() {
        UIImpactFeedbackGenerator(style: .soft).impactOccurred()
    }

    static func selection() {
        UISelectionFeedbackGenerator().selectionChanged()
    }

    static func success() {
        UINotificationFeedbackGenerator().notificationOccurred(.success)
    }
}
